packages/gateway/src/features/registration/root-resolvers.ts

```diff
@@ -415,28 +416,27 @@ export const resolvers: GQLResolver = {
         )
       }
     },
-    async fetchRecordDetailsForVerification(_, { id }, { headers }) {
+    async fetchRecordDetailsForVerification(_, { id }, context) {
       try {
         const token = await getAnonymousToken()
-        headers.Authorization = `Bearer ${token}`
+        context.headers.Authorization = `Bearer ${token}`
         const authHeader = {
-          Authorization: headers.Authorization
+          Authorization: context.headers.Authorization
         }
-        const taskEntry = await getTaskEntry(id, authHeader)
+        context.record = await getRecordById(id, authHeader.Authorization)
+
+        const taskEntry = getTaskEntryFromBundle(context.record)
 
         const taskBundle = taskBundleWithExtension(taskEntry, {
           url: 'http://opencrvs.org/specs/extension/regVerified',
-          valueString: headers['x-real-ip']!
+          valueString: context.headers['x-real-ip']!
         })
         await fetchFHIR('/Task', authHeader, 'PUT', JSON.stringify(taskBundle))
 
-        const record = await fetchFHIR(`/Composition/${id}`, authHeader)
-        if (!record) {
-          await Promise.reject(new Error('Invalid QrCode'))
-        }
-        return record
+        return context.record
       } catch (e) {
         await Promise.reject(new Error(e))
+        throw e
```

Comment:
```suggestion
        return Promise.reject(new Error('Invalid QrCode))
```

Reply:
I'm not sure if the whole block of code needs to throw at all. The "getRecordById" already throws if the record is not found, could that be sufficient?

Reply:
(correct me if I'm wrong..) Apollo hides the stack trace of the errors in production. So as a general guideline, I think it would be smart to:
1. let errors flow to Apollo through resolvers without catching them. This usually means removing try-catches.
2. if we need to handle an error cleverly in the client, lets rather throw an ApolloError, as we can then use `error.code === 'MY_CUSTOM_ERROR'` in client. Example: https://github.com/opencrvs/opencrvs-core/blob/a7ccb11f372bb2014bf382971553169b130ed7a2/packages/gateway/src/rate-limit.ts#L24